### ISSUE TITLE
The org.butterfaces.provideBootstrap=false context-param excludes ButterFaces CSS

### ISSUE THREAD

**Issue by ngriffin7a** [ButterFaces/ButterFaces#118]
Liferay Portal's out-of-the-box theme is based on Twitter Bootstrap. So for a ButterFaces portlet, it makes sense to specify the following option in WEB-INF/web.xml:
```
<context-param>
	<param-name>org.butterfaces.provideBootstrap</param-name>
	<param-value>false</param-value>
</context-param>
```
However, this seems to cause CSS styling unique to ButterFaces to not be loaded on the page. For example, when `<b:text required="true" />` fails validation, the tool tip associated with the validation failure is not displayed.

Would it be possible to separate the CSS specific to ButterFaces into a separate resource?

**Comment by larmic** [comment]
Normally butterfaces provides a smaller css if bootstrap is disabled. This seems to be an issue. I will check this.

**Comment by ngriffin7a** [comment]
@larmic We hope to have an updated release of Liferay Faces soon with support for ButterFaces. Would it be possible for you to fix this issue soon and do a patch release? Thanks.

**Comment by larmic** [comment]
Ok, when I'm using
```
<context-param>
      <param-name>org.butterfaces.provideBootstrap</param-name>
      <param-value>false</param-value>
   </context-param>
   <context-param>
      <param-name>org.butterfaces.provideJQuery</param-name>
      <param-value>false</param-value>
   </context-param>
```

the following resources will be included

```
<link type="text/css" rel="stylesheet" href="/tkkb-ui-cdi/javax.faces.resource/dist-butterfaces-only.min.css.jsf?ln=butterfaces-dist-css&amp;preventCache=1501485421641">
<script type="text/javascript" src="/tkkb-ui-cdi/javax.faces.resource/butterfaces-all-bundle.min.js.jsf?ln=butterfaces-dist-bundle-js&amp;preventCache=1501485421641"></script>
```
This seems to be correct and works in another project.

**Comment by szerhusenBC** [comment]
Hi @ngriffin7a,

we tried to reconstruct this problem in the Pluto app you provided in #115. We can't reproduced the behavior you described here. But we found another problem, the context-params Lars described above are fully ignored and every time the full Butterfaces resources are linked in the portlet's HTML.

Is it possible for you to provide a simple Maven project (preferably in Github) that just addresses the problem you're describing here?

**Comment by ngriffin7a** [comment]
@szerhusenBC Thanks so much for trying to reproduce the problem. You should be able to reproduce it with the Pluto app that I provided in #115. However, I realize that I did not provide adequate steps to reproduce the problem. Try this:

1. Edit the tomcat-7.0.42/webapps/com.liferay.faces.demo.butterfaces.applicant.portlet/WEB-INF/web.xml file and add the following:
```	<context-param>
		<param-name>org.butterfaces.provideBootstrap</param-name>
		<param-value>false</param-value>
	</context-param>```

2. Start the Pluto+Tomcat bundle

3. Visit this URL with Chrome:
http://localhost:9080/pluto/portal/butterfaces-applicant

4. Enable the JS Console in chrome

5. Click inside the "Email Address" field and enter an invalid value that does not include an "@" character such as "asdf"

6. Press the TAB key in order to cause an blur/change event.

If the bug still exists, then the following output will appear in the JS Console:
Uncaught TypeError: n.popover is not a function
    at HTMLDivElement.<anonymous> (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:26)
    at Function.each (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:1)
    at oe.fn.init.each (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:1)
    at oe.fn.init.t.fn.butterTooltip (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:26)
    at oe.fn.init.t.fn._butterTooltip (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:26)
    at HTMLDocument.<anonymous> (<anonymous>:2:124)
    at l (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:2)
    at Object.add [as done] (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:2)
    at oe.fn.init.oe.fn.ready (butterfaces-applicant?p_p_id=1_WAR_comliferayfacesdemobutterfacesapplicantportlet_INSTANCE_ABCD&p_p…:2)
    at <anonymous>:1:18

**Comment by ngriffin7a** [comment]
@szerhusenBC Note that I may have mistakenly titled this as a CSS issue because I could not see the ToolTip for the invalid Email Address field. But the JS Console might indicate that this is really a JS problem, not a CSS problem.

**Comment by ngriffin7a** [comment]
@szerhusenBC Just wondering if you had an opportunity to try and reproduce the issue. We are planning on releasing an updated version of all Liferay Faces dependencies and artifacts, and it would be great if this issue was fixed in time for that. If you are able to release ButterFaces 2.1.21 soon, then we can upgrade via [FACES-3161](https://issues.liferay.com/browse/FACES-3161). Thanks. :)

**Comment by larmic** [comment]
@ngriffin7a we've change context param in your demo application but nothing changed. we are not sure if this is a behavior of portlets?

I will talked to @szerhusenBC  within the next days.

**Comment by szerhusenBC** [comment]
@ngriffin7a I followed your instructions and now I can reproduce this problem. I will take a closer look tomorrow. At this time I can't say how complex the changes will be.

**Comment by ngriffin7a** [comment]
@szerhusenBC You are exactly right -- Liferay Portal includes Bootstrap CSS by default, but not Bootstrap JS. When I added the following to our butterfaces-applicant-portlet demo, the problem was fixed:
```
	<h:head>
		<c:if test="#{product.LIFERAY_PORTAL.detected}">
			<h:outputScript library="butterfaces-dist-bower" name="bootstrap.js" />
		</c:if>
	</h:head>
```
Thanks for your help, I will close this issue now.

**Comment by ngriffin7a** [comment]
@szerhusenBC  Will you be able to release ButterFaces 2.1.21 soon? If so, then we can include the butterfaces-applicant-portlet with our upcoming release of Liferay Faces.

**Comment by larmic** [comment]
@ngriffin7a I've released 2.1.21 and it will be available in maven central within the next hours.

**Comment by ngriffin7a** [comment]
@larmic Thanks, I fixed [FACES-3161](https://issues.liferay.com/browse/FACES-3161) and the butterfaces-applicant-portlet (and butterfaces-portlet-archetype) will be released with the upcoming version of Liferay Faces. :)
